src/Data/Common/UrlState/useUrlStateWithCurrentPage.ts

```diff
@@ -0,0 +1,50 @@
+import { isEqual, isArray } from "lodash";
+import { provide, Location, Replace, StateConfig, Update } from "./helpers";
+import { handleUrlState } from "./useUrlState";
+
+export const useUrlStateWithCurrentPage = provide(
```

Comment:
Can you add comments to this, about the logic and usage?

Reply:
Added about the wrapper, and if someone will go to this file then also will get the comment bellow about the actual handler of CurrentPage state